internal/providercache/package_install.go

```diff
@@ -157,7 +157,7 @@ func installFromLocalDir(ctx context.Context, meta getproviders.PackageMeta, tar
 
 	parentDir := filepath.Dir(absNew)
 	err = os.MkdirAll(parentDir, 0755)
-	if err != nil && os.IsExist(err) {
+	if err != nil {
 		return nil, fmt.Errorf("failed to create parent directories leading to %s: %s", targetDir, err)
```

Comment:
This additional check didn't look right to me. There's no further checking of `err` below this, and I can't see any way for `MkdirAll` to fail with an `IsExist` error—if a directory exists at any step of the given path, it returns `nil`.